kernels/rviv/3/bli_czgemm_rviv_asm_4vx4.inc

```diff
@@ -0,0 +1,768 @@
+    .text
+    .align      2
+    .global     REALNAME
+
+// void REALNAME(intptr_t k, void* alpha, void* a, void* b,
+//               void* beta, void* c, intptr_t rs_c, intptr_t cs_c)
+//
+// register arguments:
+// a0   k
+// a1   alpha
+// a2   a
+// a3   b
+// a4   beta
+// a5   c
+// a6   rs_c
+// a7   cs_c
+//
+// Assume rs_c = 1
+
+#define REALSIZE (DATASIZE/2)
+
+#define loop_counter a0
+
+#define A00_ptr   a2
+#define A10_ptr   t0
+#define A01_ptr   t1
+#define A11_ptr   t2
+
+#define B_row_ptr a3
+
+#define C00_ptr   a5
+#define C01_ptr   t3
+#define C02_ptr   t4
+#define C03_ptr   t5
+#define C10_ptr   s1
+#define C11_ptr   s2
+#define C12_ptr   s3
+#define C13_ptr   s4
+
+#define tmp       t6
+
+#define ALPHA_re  fa0
+#define ALPHA_im  fa1
+#define BETA_re   fa2
+#define BETA_im   fa3
+
+#define B00_re    fa4
+#define B00_im    fa5
+#define B01_re    fa6
+#define B01_im    fa7
+#define B02_re    fa0
+#define B02_im    fa1
+#define B03_re    fa2
+#define B03_im    fa3
+
+#define B10_re    ft0
+#define B10_im    ft1
+#define B11_re    ft2
+#define B11_im    ft3
+#define B12_re    ft4
+#define B12_im    ft5
+#define B13_re    ft6
+#define B13_im    ft7
+
+#define A00_re    v24
+#define A00_im    v25
+#define A10_re    v26
+#define A10_im    v27
+#define A01_re    v28
+#define A01_im    v29
+#define A11_re    v30
+#define A11_im    v31
+
+#define C0_re     v24
+#define C0_im     v25
+#define C1_re     v26
+#define C1_im     v27
+#define C2_re     v28
+#define C2_im     v29
+#define C3_re     v30
+#define C3_im     v31
+
+#define AB00_re   v0
+#define AB00_im   v1
+#define AB01_re   v2
+#define AB01_im   v3
+#define AB02_re   v4
+#define AB02_im   v5
+#define AB03_re   v6
+#define AB03_im   v7
+#define AB10_re   v8
+#define AB10_im   v9
+#define AB11_re   v10
+#define AB11_im   v11
+#define AB12_re   v12
+#define AB12_im   v13
+#define AB13_re   v14
+#define AB13_im   v15
+
+#define tmp0_re   v16
+#define tmp0_im   v17
+#define tmp1_re   v18
+#define tmp1_im   v19
+#define tmp2_re   v20
+#define tmp2_im   v21
+#define tmp3_re   v22
+#define tmp3_im   v23
+
+#define rs_c  a6
+#define cs_c  a7
+
+REALNAME:
+    #include "rviv_save_registers.inc"
+
+    // Arbitrarily long length
+    // TODO: Find a better way of querying MAXVL
+    li tmp, 256
+    vsetvli s0, tmp, VTYPE, m1, ta, ma
+    slli s0, s0, DATASHIFT      // vector length in bytes
+
+    // Workaround for `mul rs_c, rs_c, s0` because rv64iv does not support `mul`
+    mv rs_c, s0                 // Assumes unit row skip
```

Comment:
There is an inconsistency here. This command overwrites `rs_c` and, hence, renders the caller's value `rs_c * num_fp32_per_vector()` superfluous.

Reply:
I am moving all of the calculations to the C code, which makes it simpler. The strides we pass to the assembly are measured in bytes and do not need to be multiplied or shifted. The `DATASHIFT` macro is eliminated.

Reply:
:+1: